app/src/main/java/com/example/sharingang/AccountFragment.kt

```diff
@@ -0,0 +1,101 @@
+package com.example.sharingang
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.databinding.DataBindingUtil
+import androidx.fragment.app.Fragment
+import com.example.sharingang.databinding.FragmentAccountBinding
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInClient
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import com.google.android.gms.common.api.ApiException
+import com.google.android.gms.tasks.OnCompleteListener
+
+
+class AccountFragment : Fragment() {
+
+    private val CLIENT_AUTH_KEY: String =
+        "771023799063-kmve17s9cfu6ckd3kijcv8vdvvdqb58s.apps.googleusercontent.com"
```

Comment:
Should the key be here in the file?

Reply:
The implementations I have seen using it do the same as me. The SHA1 key for authenticating the app is put in oAuth for the project, therefore it is fine to use this OAuth-generated key here as an attribute. Is there a better place you would put it?

Reply:
Isn't such a key supposed to be private? (aka not available directly on the repo?)